File: src/data/scripts/campaign/econ/niko_MPC_overgrownNanoforge.java
package data.scripts.campaign.econ;

import com.fs.starfarer.api.impl.campaign.econ.BaseHazardCondition;
import data.utilities.niko_MPC_ids;

public class niko_MPC_overgrownNanoforge extends BaseHazardCondition {

    @Override
    public void apply(String id) {
        if (!market.hasIndustry(niko_MPC_ids.overgrownNanoforgeIndustryId)) {
            addIndustryToMarket();
        }
        super.apply(id);
    }

    private void addIndustryToMarket() {
        market.addIndustry(niko_MPC_ids.overgrownNanoforgeIndustryId);
    }
}

File: src/data/scripts/campaign/econ/conditions/overgrownNanoforge/Niko_MPC_overgrownNanoforge.kt
package data.scripts.campaign.econ.conditions.overgrownNanoforge

import data.scripts.campaign.econ.conditions.niko_MPC_industryAddingCondition
import data.utilities.niko_MPC_industryIds

class niko_MPC_overgrownNanoforge : niko_MPC_industryAddingCondition() {

    init {
        industryIds.add(niko_MPC_industryIds.overgrownNanoforgeIndustryId)

        //todo: add structure ids and override fun from super and shit
    }

}